src/cgmemmgr.cpp

```diff
@@ -0,0 +1,860 @@
+    // file system.
+#  ifdef __NR_memfd_create
+    fd = syscall(__NR_memfd_create, "julia-codegen", MFD_CLOEXEC);
+    if (check_fd_or_close(fd))
+        return fd;
+#  endif
+#  ifdef _OS_FREEBSD_
+    fd = shm_open(SHM_ANON, O_RDWR, S_IRWXU);
+    if (check_fd_or_close(fd))
+        return fd;
+#  endif
+    char shm_name[] = "julia-codegen-0123456789-0123456789/tmp///";
+    pid_t pid = getpid();
+    // `shm_open` can't be mapped exec on mac
+#  ifndef _OS_DARWIN_
+    do {
+        snprintf(shm_name, sizeof(shm_name),
+                 "julia-codegen-%d-%d", (int)pid, rand());
+        fd = shm_open(shm_name, O_RDWR | O_CREAT | O_EXCL, S_IRWXU);
+        if (check_fd_or_close(fd)) {
+            shm_unlink(shm_name);
+            return fd;
+        }
+    } while (errno == EEXIST);
+#  endif
+    FILE *tmpf = tmpfile();
+    if (tmpf) {
+        fd = dup(fileno(tmpf));
+        fclose(tmpf);
+        if (check_fd_or_close(fd)) {
+            return fd;
+        }
+    }
+    snprintf(shm_name, sizeof(shm_name),
+             "/tmp/julia-codegen-%d-XXXXXX", (int)pid);
+    fd = mkstemp(shm_name);
+    if (check_fd_or_close(fd)) {
+        unlink(shm_name);
+        return fd;
+    }
+    return -1;
+}
+
+static size_t map_offset = 0;
+// Multiple of 128MB.
+// Hopefully no one will set a ulimit for this to be a problem...
+static constexpr size_t map_size_inc = 128 * 1024 * 1024;
+static size_t map_size = 0;
+static jl_mutex_t shared_map_lock;
+
+static void *create_shared_map(size_t size, size_t id)
+{
+    void *addr = mmap(nullptr, size, PROT_READ | PROT_WRITE, MAP_SHARED,
+                      anon_hdl, id);
+    assert(addr != MAP_FAILED && "Cannot map RW view");
+    return addr;
+}
+
+static intptr_t init_shared_map()
+{
+    anon_hdl = get_anon_hdl();
+    if (anon_hdl == -1)
+        return -1;
+    map_offset = 0;
+    map_size = map_size_inc;
+    int ret = ftruncate(anon_hdl, map_size);
+    if (ret != 0) {
+        perror(__func__);
+        abort();
+    }
+    return anon_hdl;
+}
+
+static void *alloc_shared_page(size_t size, size_t *id, bool exec)
+{
+    assert(size % jl_page_size == 0);
+    size_t off = jl_atomic_fetch_add(&map_offset, size);
+    *id = off;
+    if (__unlikely(off + size > map_size)) {
+        JL_LOCK_NOGC(&shared_map_lock);
+        size_t old_size = map_size;
+        while (off + size > map_size)
+            map_size += map_size_inc;
+        if (old_size != map_size) {
+            int ret = ftruncate(anon_hdl, map_size);
+            if (ret != 0) {
+                perror(__func__);
+                abort();
+            }
+        }
+        JL_UNLOCK_NOGC(&shared_map_lock);
+    }
+    return create_shared_map(size, off);
+}
+#endif // _OS_WINDOWS_
+
+#ifdef _OS_LINUX_
+// Using `/proc/self/mem`, A.K.A. Keno's remote memory manager.
+
+static int self_mem_fd = -1;
+
+static int init_self_mem()
+{
+    int fd = open("/proc/self/mem", O_RDWR | O_SYNC | O_CLOEXEC);
```

Comment:
That kernel is actually too old for this method to work although I guess we should still compile this method for this to be included in the generic binary.... Should be fixed in https://github.com/JuliaLang/julia/commit/3f4e1e7274993b32b4dab8bdabdc9b645993923c